docs/metrics-data-model.md

```diff
@@ -0,0 +1,177 @@
+# k6 Time Series
+
+We would migrate to a time series data model for getting the benefits of aggregated data so just allocating samples one time and pass around the aggregated time series value (or values - e.g. in case of Trend). The Outputs will switch from fetching the buffered samples to fetching a snapshot of the time series (the same concept adopted by Prometheus' scraping operation).
+
+## Data model
+
+#### Metrics and Series
+
+* Metric (name, type)
+* TimeSeries (metric, tagset)
+* Sample (metric, tagset, timestamp, value)
+* TimeSeries Sample (timeseries ref, timestamp, value)
+
+```go
+// This is the current k6 Metric polished from the Thresholds stuff.
+type Metric struct {
+	Name     string     `json:"name"`
+	Type     MetricType `json:"type"`
+	Contains ValueType  `json:"contains"`
+}
+
+type TimeSeries struct {
+    ID uin64 // hash(metric+tags)
+	MetricName string
+	Tags   TagSet
+
+	Sink Sink // Counter, Guage, Rate, Trend (SparseHistogram in the future)
+}
+
+type Sink interface {
+    AddPoint(v float64)
+}
+
+type TSample struct { 
+	TimeSeriesID string
+	Timestamp      uint64
+	Value          float64
+}
```

Comment:
```suggestion
```
It isn't really required, the engine / ingester will resolve the time series then it invokes `AddPoint(Sample.Value)`. Not really required to have a new struct.